common/src/edgeai_demo.cpp

```diff
@@ -0,0 +1,1147 @@
+            graph_parameters_queue_params_list[graph_parameter_index].refs_list_size = 1;
+            graph_parameters_queue_params_list[graph_parameter_index].refs_list = (vx_reference*)&m_imgMosaicObjs[i]->imgMosaicObj.output_image[0];
+            graph_parameter_index++;
+        }
+
+        if((vx_status)VX_SUCCESS == status)
+        {
+            /* Background image index of Mosaic Node is 2 */
+            status = add_graph_parameter_by_node_index(m_ovxGraph->graph, m_imgMosaicObjs[i]->imgMosaicObj.node, 2);
+            m_imgMosaicObjs[i]->imgMosaicObj.background_graph_parameter_index = graph_parameter_index;
+            graph_parameters_queue_params_list[graph_parameter_index].graph_parameter_index = graph_parameter_index;
+            graph_parameters_queue_params_list[graph_parameter_index].refs_list_size = 1;
+            graph_parameters_queue_params_list[graph_parameter_index].refs_list = (vx_reference*)&m_imgMosaicObjs[i]->imgMosaicObj.background_image[0];
+            graph_parameter_index++;
+        }
+
+    }
+
+    /* Schedule graph */
+    if((vx_status)VX_SUCCESS == status)
+    {
+        /* Scheduling graph in auto mode. */
+        status = vxSetGraphScheduleConfig(m_ovxGraph->graph,
+                    VX_GRAPH_SCHEDULE_MODE_QUEUE_AUTO,
+                    graph_parameter_index,
+                    graph_parameters_queue_params_list);
+    }
+
+    /* Verify graph*/
+    if((vx_status)VX_SUCCESS == status)
+    {
+        status = vxVerifyGraph(m_ovxGraph->graph);
+    }
+
+    if(m_cameraObj != NULL)
+    {
+        if (m_cameraObj->captureObj.enable_error_detection && status == VX_SUCCESS)
+        {
+            status = tiovx_capture_module_send_error_frame(&m_cameraObj->captureObj);
+        }
+    }
+
+    for(uint i=0; i < m_multiScalerObjs.size(); i++)
+    {
+        tiovx_multi_scaler_module_update_filter_coeffs(&m_multiScalerObjs[i]->multiScalerObj1);
+        tiovx_multi_scaler_module_update_crop_params(&m_multiScalerObjs[i]->multiScalerObj1);
+        if (m_multiScalerObjs[i]->useSecondaryMsc)
+        {
+            tiovx_multi_scaler_module_update_filter_coeffs(&m_multiScalerObjs[i]->multiScalerObj2);
+            tiovx_multi_scaler_module_update_crop_params(&m_multiScalerObjs[i]->multiScalerObj2);
+        }
+    }
+
+    return status;
+}
+
+int32_t EdgeAIDemoImpl::startDemo()
+{
+    int                 status = 0;
+    uint32_t            num_refs;
+    vx_image            input_o, output_o, background_o;
+    vx_object_array     input_obj_arr;
+    bool                camera_first_deq_done = false;
+    uint32_t            cnt = 0;
+
+    if(m_cameraObj != NULL)
+    {
+        /* Start sensor module. */
+        status = tiovx_sensor_module_start(&m_cameraObj->sensorObj);
+        for(int buf_id = 0; buf_id < m_cameraObj->captureObj.out_bufq_depth; buf_id++)
+        {
+            vxGraphParameterEnqueueReadyRef(m_ovxGraph->graph,
```

Comment:
Should return status and check.

Reply:
Added status protection here. Updated the PR